docs/cep/source/stream-source.md

```diff
@@ -0,0 +1,183 @@
+    "stock":{
+      "volume":100,
+      "company":{
+        "symbol":"FB"
+      },
+      "price":55.6
+    }
+  }
+}
+```
+
+The configuration of the `database` source and the custom `JSON` source mapping to achieve the above is as follows.
+
+```sql
+CREATE SOURCE StockStream WITH (source.type='database', collection='foo', map.type='json', enclosing.element="$.portfolio",) (symbol = "stock.company.symbol", price = "stock.price", volume = "stock.volume");
+```
+
+The same can also be configured by omitting the attribute names as below.
+
+```sql
+CREATE SOURCE StockStream WITH (source.type='database', collection='foo', map.type='json', enclosing.element="$.portfolio",) ("stock.company.symbol", "stock.price", "stock.volume");
+```
+
+## Error Handling at Stream Source
+
+When errors are thrown by stream elements subscribed to the stream, the error gets propagated up to the stream that delivered the event to those stream elements. By default the error is logged and dropped at the stream, but this behavior can be altered by by adding `OnError` property to the corresponding stream definition.
+
+### OnError Property
+
+`OnError` property can help users to capture the error and the associated event, and handle them gracefully by sending them to a fault stream.
+
+The `OnError` property and the required `action` to be specified as below.
+
+```sql
+CREATE SOURCE <stream name> WITH (OnError.action='<action>') (<attribute name> <attribute type>, <attribute name> <attribute type>, ... );
+```
+
+The `action` parameter of the `OnError` property defines the action to be executed during failure scenarios.
+
+The following actions can be specified to `OnError` property to handle erroneous scenarios.
+
+- `STREAM`: Creates a fault stream and redirects the event and the error to it. The created fault stream will have all the attributes defined in the base stream to capture the error causing event, and in addition it also contains `_error` attribute of type `object` to containing the error information. The fault stream can be referred by adding `!` in front of the base stream name as `!<stream name>`.
+
+### Example
+
+Handle errors in `TempStream` by redirecting the errors to a fault stream.
+
+The configuration of `TempStream` stream and `OnError` property is as follows:
+
+```sql
+CREATE STREAM TempStream WITH(OnError.action="STREAM") (deviceID long, roomNo int, temp double;
+```
+
+Stream infers and automatically defines the fault stream of `TempStream` as given below.
+
+```sql
+CREATE STREAM !TempStream (deviceID long, roomNo int, temp double, _error object);
+```
+
+The stream worker extends the above use case by adding failure generation and error handling with the use of queries is as follows.
+
+```sql
+-- Define fault stream to handle error occurred at TempStream subscribers
+
+CREATE STREAM TempStream WITH(OnError.action="STREAM") (deviceID long, roomNo int, temp double;
+
+-- Error generation through a custom function `createError()`
```

Comment:
@diana-macrometa - Let' take this out, we don't have such custom function.

Reply:
No, this is not a JS function. The description is for a custom extension, so called `stream processor`.

Reply:
Did I remove the wrong part? I thought I removed the custom function bit you were referring to.